test/hotspot/jtreg/compiler/codecache/stress/OverloadCompileQueueTest.java

```diff
@@ -77,15 +102,16 @@
         }
     }
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws InterruptedException {
+        lockUnlockThread.start();
+
         if (Platform.isInt()) {
             throw new Error("TESTBUG: test can not be run in interpreter");
         }
         new CodeCacheStressRunner(new OverloadCompileQueueTest()).runTest();
-    }
 
-    public OverloadCompileQueueTest() {
-        Helper.startInfiniteLoopThread(this::lockUnlock, 100L);
+        lockUnlockThread.isActive = false;
+        lockUnlockThread.join();
```

Comment:
So this now relies on `lockUnlockThread` unblocking from its `Thread.sleep`-s, and then `joining` here? We could wait here for up to `MAX_SLEEP` seconds then? Maybe we should command `lockUnlockThread.interrupt()` before `join()` to make the test a tad faster?

Reply:
Added an interrupt, thanks.